cmd/lifecycle/rebaser.go

```diff
@@ -79,32 +92,72 @@ func (r *rebaseCmd) Privileges() error {
 }
 
 func (r *rebaseCmd) Exec() error {
+	var err error
+	var newBaseImage imgutil.Image
+	if r.useDaemon {
+		newBaseImage, err = local.NewImage(
+			r.imageNames[0],
+			r.docker,
+			local.FromBaseImage(r.runImageRef),
+		)
+	} else {
+		newBaseImage, err = remote.NewImage(
+			r.imageNames[0],
+			r.keychain,
+			remote.FromBaseImage(r.runImageRef),
+		)
+	}
+	if err != nil || !newBaseImage.Found() {
+		return cmd.FailErr(err, "access run image")
+	}
+
+	rebaser := &lifecycle.Rebaser{
+		Logger: cmd.DefaultLogger,
+	}
+	report, err := rebaser.Rebase(r.appImage, newBaseImage, r.imageNames[1:])
+	if err != nil {
+		return cmd.FailErrCode(err, cmd.CodeRebaseError, "rebase")
+	}
+	if err := lifecycle.WriteTOML(r.reportPath, &report); err != nil {
+		return cmd.FailErrCode(err, cmd.CodeRebaseError, "write rebase report")
+	}
+	return nil
+}
+
+func (r *rebaseCmd) registryImages() []string {
+	registryImages := r.imageNames
+	if r.runImageRef != "" {
+		registryImages = append(registryImages, r.runImageRef)
```

Comment:
This is another case where we need to resolve the best mirror from the stack metadata first. I see that we are actually doing this because we call `setAppImage` first. However `setAppImage` actually needs the `keychain` to be initialized.